godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/AABB.kt

```diff
@@ -0,0 +1,471 @@
+                over = true
+            else
+                under = true
+        }
+
+        return under && over
+    }
+
+    /**
+     * Returns true if the AABB intersects the line segment between from and to.
+     */
+    fun intersectsSegment(from: Vector3, to: Vector3): Boolean {
+        var min = 0.0
+        var max = 0.0
+
+        for (i in 0..2) {
+            val segFrom = from[i]
+            val segTo = to[i]
+            val boxBegin = position[i]
+            val boxEnd = boxBegin + size[i]
+            val cmin: RealT
+            val cmax: RealT
+
+            if (segFrom < segTo) {
+                if (segFrom > boxEnd || segTo < boxBegin) {
+                    return false
+                }
+                val length = segTo - segFrom
+                cmin = if (segFrom < boxBegin) ((boxBegin - segFrom) / length) else 0.0
+                cmax = if (segTo > boxEnd) ((boxEnd - segFrom) / length) else 1.0
+            } else {
+                if (segTo > boxEnd || segFrom < boxBegin) {
+                    return false
+                }
+                val length = segTo - segFrom
+                cmin = if (segFrom > boxEnd) (boxEnd - segFrom) / length else 0.0
+                cmax = if (segTo < boxBegin) (boxBegin - segFrom) / length else 1.0
+            }
+            if (cmin > min) {
+                min = cmin
+            }
+            if (cmax < max)
+                max = cmax
+            if (max < min) {
+                return false
+            }
+        }
+        return true
+    }
+
+    /**
+     * Returns true if this AABB and aabb are approximately equal, by running isEqualApprox on each component.
+     */
+    fun isEqualApprox(other: AABB): Boolean {
+        return this.position.isEqualApprox(other.position) && this.size.isEqualApprox(other.size)
+    }
+
+    /**
+     * Returns a larger AABB that contains both this AABB and with.
+     */
+    fun merge(p_with: AABB): AABB {
+        val aabb = this
+        aabb.mergeWith(p_with)
+        return aabb
+    }
+
+    internal fun mergeWith(other: AABB) {
+        val beg1 = position
+        val beg2 = other.position
+        val end1 = position + size
+        val end2 = other.position + other.size
+
+        val min = Vector3()
+        val max = Vector3()
+
+        min.x = if (beg1.x < beg2.x) beg1.x else beg2.x
+        min.y = if (beg1.y < beg2.y) beg1.y else beg2.y
+        min.z = if (beg1.z < beg2.z) beg1.z else beg2.z
+
+        max.x = if (end1.x > end2.x) end1.x else end2.x
+        max.y = if (end1.y > end2.y) end1.y else end2.y
+        max.z = if (end1.z > end2.z) end1.z else end2.z
+
+        position = min
+        size = max - min
+    }
+
+    //UTILITIES
+    override fun equals(other: Any?): Boolean =
+        when (other) {
+            is AABB -> (position == other.position && size == other.size)
+            else -> false
+        }
+
+    override fun toString(): String =
+        "$position - $size"
```

Comment:
Can we make this `AABB(position=<position>, size=<size>)`? I find it more easier when debugging when the class name and its properties is included.

Reply:
![image](https://user-images.githubusercontent.com/31470327/81495331-dd6c0f00-92af-11ea-8657-ad16c6e18fd5.png)

I'm using the toString format of the Cpp object.  We can do our own thing if you wish.

Reply:
I agree with @raniejade here.